ext/mysqli/mysqli_api.c

```diff
@@ -729,14 +731,19 @@ PHP_FUNCTION(mysqli_data_seek)
 		RETURN_THROWS();
 	}
 
+	if (offset < 0) {
+		zend_argument_value_error(ERROR_ARG_POS(2), "must be greater than or equal to 0");
+		RETURN_THROWS();
+	}
+
 	MYSQLI_FETCH_RESOURCE(result, MYSQL_RES *, mysql_result, "mysqli_result", MYSQLI_STATUS_VALID);
 
 	if (mysqli_result_is_unbuffered(result)) {
-		php_error_docref(NULL, E_WARNING, "Function cannot be used with MYSQL_USE_RESULT");
-		RETURN_FALSE;
+		zend_throw_error(NULL, "Function cannot be used with MYSQL_USE_RESULT");
```

Comment:
```suggestion
		zend_throw_error(NULL, "mysqli_data_seek() cannot be used with MYSQL_USE_RESULT");
```
etc.

Reply:
I have suggested a PR for this error. I believe the constant should be MYSQLI_USE_RESULT

Reply:
Right, this should be MYSQLI_USE_RESULT.

Reply:
I still think the message I suggested is better "cannot be used in MYSQLI_USE_RESULT mode". The constant describes result mode. You don't use the function with this constant but rather your result is in one or the other mode.

Reply:
You can address this in your PR, I just want to get done with this one at this point ^^"